File: algo-deaf-grandma/deaf_grandma.js
function deafGrandma() {
  let reply
  let goodbyeCount = 0
  let convo = prompt("Say something to Grandma?")
  
  while (convo != null) {
    if (convo === "") {
      reply = "WHAT?!"
    } else if (convo === "GOODBYE!") {
      if (goodbyeCount === 0) {
        reply = "LEAVING SO SOON?"
        goodbyeCount++
      } else {
        reply = "LATER, SKATER!"
        alert(`You said: ${convo} \nGrandma replied: ${reply}`)
        document.writeln("Program exited...")
        return
      }
    } else if (convo.toUpperCase() === convo) {
      reply = "NO, NOT SINCE 1946!"
    } else reply = "SPEAK UP, KID!"
    
    alert(`You said: ${convo} \nGrandma replied: ${reply}`)
    convo = prompt("Say something to Grandma?")
  }
  document.writeln("User selected to cancel. Program exited...")
  return
}

deafGrandma()

File: algo-deaf-grandma/deaf_grandma.py
def deaf_grandma():
    exit = False
    previous_goodbye = False
    
    print('You have engaged into a conversation with your granny.')
    print('You can just stop talking and walk away by saying, "I\'m out!"\n')
    
    while(exit == False):
        talk = input('Say something to granny: ')
       
        if talk == "":
            print('WHAT?!')
        elif talk.lower() == "i'm out!" or talk.lower() == "i'm out":
            exit = True
        elif talk == 'GOODBYE!':
            if previous_goodbye:
                print('LATER, SKATER!')
                break                
            else:
                previous_goodbye = True
                print('LEAVING SO SOON?')
        elif talk.isupper():
            print('NO, NOT SINCE 1946!')
        else:
            print('SPEAK UP, KID!')

deaf_grandma()